### PR TITLE
TASK: Improve StringHelper documentation

### PR DESCRIPTION
**Pull request by hlubek**
Adds missing PHPDoc to StringHelper functions and add more examples.

### CODE REVIEW

**Comment by hlubek** [comment]
StyleCI failing is not caused by this commit, I'd rather not want to fix this here because of possible merge errors.

File: TYPO3.Eel/Classes/TYPO3/Eel/Helper/StringHelper.php [hunk 1]
```diff
@@ -555,4 +574,5 @@ public function allowsCallOfMethod($methodName)
     {
         return true;
     }
+
```
Comment (albe) [inline TYPO3.Eel/Classes/TYPO3/Eel/Helper/StringHelper.php]: This addition of a line break is the reason for the Style-CI error, can easily be fixed ;)

Comment (hlubek) [inline TYPO3.Eel/Classes/TYPO3/Eel/Helper/StringHelper.php]: Ah okay, didn't see that, I'll push a fix

**Comment by hlubek** [comment]
Should be fixed.

**Pull request merged**